addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,130 @@
+import requests
+
+from odoo import models, fields, _, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_ro_edi_document_ids = fields.One2many(
+        comodel_name='l10n_ro_edi.document',
+        inverse_name='invoice_id',
+    )
+    l10n_ro_edi_active_document_id = fields.Many2one(
+        comodel_name='l10n_ro_edi.document',
+        compute='_compute_from_l10n_ro_edi_document_ids',
+    )
+    l10n_ro_edi_state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sending_failed', 'Error'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        compute='_compute_from_l10n_ro_edi_document_ids_stored',
+        store=True,
+    )
+    l10n_ro_edi_message = fields.Char(compute='_compute_from_l10n_ro_edi_document_ids')
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_active_document_id = active_document
+            move.l10n_ro_edi_message = active_document.message
+
+    @api.depends('l10n_ro_edi_document_ids')
+    def _compute_from_l10n_ro_edi_document_ids_stored(self):
+        for move in self:
+            active_document = self.env['l10n_ro_edi.document'].search([('invoice_id', '=', move.id)], limit=1)
+            move.l10n_ro_edi_state = active_document.state
```

Comment:
```suggestion
@api.depends('l10n_ro_edi_document_ids')
def _compute_l10n_ro_edi_state(self):
    for move in self:
        move.l10n_ro_edi_state = None
        for document in move.l10n_ro_edi_document_ids.sorted():
            if document.state in ('invoice_sending', 'invoice_sent'):
                move.l10n_ro_edi_state = document.state
                break
```

`l10n_ro_edi_active_document_id` looks useless. Imo, you just need to copy the valuable data to the invoice and let the document acting as logs.
`l10n_ro_edi_message` is useless too: if you want to get more details about what failed, go to the documents. That's the purpose of such model.